File: java/src/main/java/com/mertsy/javasample/CapturingActivity.java
package com.mertsy.javasample;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.os.Bundle;

import com.mertsy.capturing.CapturingResultListener;
import com.mertsy.capturing.CapturingType;
import com.mertsy.capturing.MertsyCapturingView;

public class CapturingActivity extends BaseActivity {

    public static final String TAG = "CapturingActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_capturing);

        CapturingType type = getIntent().getParcelableExtra(TAG);

        MertsyCapturingView capturingView = findViewById(R.id.capturingView);

        //Or inside xml app:mertsyCapturingType
        if (type != null)
            capturingView.setCapturingType(type);

        capturingView.setLifecycleOwner(this);

        capturingView.setFragmentManager(getSupportFragmentManager());

        capturingView.setBackPressedDispatcher(getOnBackPressedDispatcher());

        capturingView.setCapturingResultListener(new CapturingResultListener() {
            @Override
            public void onComplete(String modelId) {
                ClipboardManager clipboard = (ClipboardManager) getSystemService(Context.CLIPBOARD_SERVICE);
                ClipData clip = ClipData.newPlainText("modelId", modelId);
                clipboard.setPrimaryClip(clip);
                showToast("Copied\n" + modelId);
                finish();
            }

            @Override
            public void onCancel() {
                finish();
            }
        });

    }

}

File: kotlin/src/main/java/com/mertsy/kotlinsample/CapturingActivity.kt
package com.mertsy.kotlinsample

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.mertsy.capturing.CapturingType
import com.mertsy.capturing.MertsyCapturingView

class CapturingActivity: AppCompatActivity(R.layout.activity_capturing) {

    companion object {
        const val TAG = "CapturingActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val type: CapturingType? = intent.getParcelableExtra(TAG)

        with(findViewById<MertsyCapturingView>(R.id.capturingView)) {

            //Or inside xml xml app:mertsyCapturingType
            type?.let { setCapturingType(it) }

            setLifecycleOwner(this@CapturingActivity)

            setFragmentManager(supportFragmentManager)

            setBackPressedDispatcher(onBackPressedDispatcher)

            setCapturingResultCallback(onComplete = { modelId: String ->
                context.copyToClipboard(modelId)
                showToast("Copied\n$modelId")
                finish()
            }, onCancel = ::finish)

            /*
            *
            * OR
            *
            * setCapturingResultCallback(onComplete = {}, onCancel = {})
            *
            * */
        }
    }
    
}